### PR TITLE
bugfix(nix-shell/happDev): compile hc with `--release`

### PR DESCRIPTION
**Pull request by steveeJ**
this bug became apparent since d902cde90f3b727b9bf5c957dfb287f1f211c717.
it caused a runtime bug in `hc` related to CLI flag handling in one of
the dependencies. most likely it's a bug in structopt or clap dependency.

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: nix/pkgs/default.nix [hunk 1]
```diff
@@ -56,15 +58,16 @@ let
...
+      inherit hcToplevelDir;
+      releaseAutomation = "${hcReleaseAutomation}/bin/hc-ra";
+    } // {
     inherit hcReleaseAutomation;
   };
   happ = {
-    holochain = mkHolochainBinaryScript "holochain";
-    hc = mkHolochainBinaryScript "hc";
+    holochain = mkHolochainBinaryScript "holochain" "";
+    hc = mkHolochainBinaryScript "hc" "--release";
```
Comment (maackle) [inline nix/pkgs/default.nix:70]: Why does `hc` get a release build but not `holochain`?

Comment (steveeJ) [reply nix/pkgs/default.nix:70]: i'd prefer if we wouldn't build `hc` with `--release` either in this case. the point is that these wrapper binaries will be used in debugging contexts locally, most likely by holochain core devs who iterate on the core code while checking how the code changes integrate into other apps they try out locally.

the point of this PR is to make `hc` work at all in this context. the issue is that its debug build currently produces broken runtime behavior. that is an issue to take care off later, this one makes it work in the meantime.

Comment (maackle) [reply nix/pkgs/default.nix:70]: I'm curious what the problem is. Can you put a comment explaining the problem here so we know that this is a temporary fix, and what will enable us to remove it?

Comment (steveeJ) [reply nix/pkgs/default.nix:70]: the commit message has some details, except the actual runtime error that occurs on debug builds. here it is:

```
+./target/debug/hc --version
thread 'main' panicked at 'Argument short must be unique

    -o is already in use', ./.cargo/registry/src/github.com-1ecc6299db9ec823/clap-2.34.0/src/app/parser.rs:190:13
```

Comment (steveeJ) [reply nix/pkgs/default.nix:70]: @maackle ping

Comment (maackle) [reply nix/pkgs/default.nix:70]: I feel like we should just fix this in `hc`, looks like an easy fix

Comment (maackle) [reply nix/pkgs/default.nix:70]: FYI I am able to run a debug build of `hc` just fine locally on develop, i.e. this works with no error:

```
cargo build --bin hc && ./target/debug/hc --version
```

Comment (steveeJ) [reply nix/pkgs/default.nix:70]: indeed! i tried it out just now and it has been (coincidentally) repaired in the meantime

**Comment by steveeJ** [comment]
this has been (coincidentally) repaired in the meantime

**Pull request closed without merging**